### PR TITLE
[thci] selectively configure Domain Prefix advertisement on the host

### PR DESCRIPTION
**Pull request by edmont**
Implement changes in `setupHost` method to be able to selectively
configure the RA for the Domain Prefix.

See https://threadgroup.atlassian.net/browse/DEV-2191 discussion.

### CODE REVIEW

File: tools/harness-thci/OpenThread_BR.py [hunk 1]
```diff
@@ -346,16 +346,15 @@ def _beforeRegisterMulticast(self, sAddr='ff04::1234:777a:1', timeout=300):
         self.bash(cmd)
 
     @API
-    def setupHost(self, setDua=False):
+    def setupHost(self, setDp=False, setDua=False):
```
Comment (simonlingoogle) [inline tools/harness-thci/OpenThread_BR.py:349]: I can see why it's fine to use `setDp=False` in non-DUA TCs. 

But are there any tests that require `setDp=False` to pass?

Comment (edmont) [reply tools/harness-thci/OpenThread_BR.py:349]: Probably not, but it was agreed to have this behavior on the reference host device, i.e., to be able to stop advertising the Domain Prefix on the backbone link.

Comment (simonlingoogle) [reply tools/harness-thci/OpenThread_BR.py:349]: SGTM. Thanks for clarification.

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Pull request merged**